src/infrastructure/Listener.ts

```diff
@@ -92,9 +94,14 @@ export class Listener implements IListener {
          * WebSocket injected when using listeners in client.
          */
         private websocketInjected?: any,
+        /**
+         * Multisig repository for resolving multisig accounts
+         */
+        private multisigRepository?: MultisigRepository,
     ) {
         this.url = url.replace(/\/$/, '');
         this.messageSubject = new Subject();
+        this.multisigRepository = this.multisigRepository ? this.multisigRepository : new MultisigHttp(this.namespaceRepository.getUrl());
```

Comment:
Why asking for the url to the node repository? The URL is up there plus I would like to keep the repository URL agnostic.

Reply:
It's kind of hacky. I would make multisigRepository required like namespaceRepository and remove getUrl from the namespace repository. If somebody is already using the ListnerHttp, this person would know how to create a multisigRepository.

Reply:
is the way not to introduce a breaking change at this point of time

Reply:
getUrl is breaking the abstraction. Also why in the NamespaceRepository and not in the other ones (I'm not saying they should have it, it's just inconsistent). 

If the devs coded the clients correctly, the construction of the Listener should be behind the scenes without breaking the code. If not, most people would know how to create the repositories used by the listener.

Those are my points, if you like to merge it's all good.